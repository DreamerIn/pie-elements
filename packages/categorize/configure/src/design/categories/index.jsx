import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import classNames from 'classnames';
import Category from './category';
import Header from '../header';

export class Categories extends React.Component {
  static propTypes = {
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    }),
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    columns: PropTypes.number.isRequired,
    onColumnsChange: PropTypes.func.isRequired,
    categories: PropTypes.array,
    onChange: PropTypes.func.isRequired,
    onDeleteChoice: PropTypes.func.isRequired,
    onAddChoice: PropTypes.func.isRequired,
    onAdd: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };

  static defaultProps = {
    columns: 2
  };

  changeCategory = c => {
    const { onChange, categories } = this.props;

    const index = categories.findIndex(a => a.id === c.id);
    if (index !== -1) {
      categories.splice(index, 1, c);
      onChange(categories);
    }
  };

  render() {
    const {
      columns,
      classes,
      className,
      onColumnsChange,
      onDeleteChoice,
      categories,
      onAdd,
      onDelete,
      onAddChoice,
      imageSupport
    } = this.props;

    const holderStyle = {
      gridTemplateColumns: `repeat(${columns}, 1fr)`
    };

    return (
      <div className={classNames(classes.categories, className)}>
        <Header label="Categories" buttonLabel="ADD A CATEGORY" onAdd={onAdd} />
        <div className={classes.row}>
          <TextField
            label="Categories per row"
            type="number"
            inputProps={{
              min: 1,
              max: 4
            }}
            value={columns}
            onChange={onColumnsChange}
          />
        </div>
        <div className={classes.categoriesHolder} style={holderStyle}>
          {categories.map((category, index) => (
            <Category
              key={index}
              imageSupport={imageSupport}
              category={category}
              onChange={this.changeCategory}
              onDelete={() => onDelete(category)}
              onDeleteChoice={(choice, choiceIndex) =>
                onDeleteChoice(category, choice, choiceIndex)
              }
              onAddChoice={onAddChoice}
            />
          ))}
        </div>
      </div>
    );
  }
}

const styles = theme => ({
  categories: {
    marginBottom: theme.spacing.unit
  },
  categoriesHolder: {
    display: 'grid',
    gridRowGap: `${theme.spacing.unit}px`,
    gridColumnGap: `${theme.spacing.unit}px`
  },
  row: {
    display: 'grid',
    gridTemplateColumns: 'repeat(2, 1fr)',
    gridColumnGap: `${theme.spacing.unit}px`,
    alignItems: 'baseline',
    width: '100%',
    marginTop: theme.spacing.unit,
    marginBottom: 2 * theme.spacing.unit
  }
});

export default withStyles(styles)(Categories);
