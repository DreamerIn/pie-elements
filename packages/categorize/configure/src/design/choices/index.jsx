import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Choice from './choice';
import Header from '../header';
import every from 'lodash/every';
import Config from './config';

export class Choices extends React.Component {
  static propTypes = {
    config: PropTypes.object.isRequired,
    onConfigChange: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    choices: PropTypes.array.isRequired,
    onChange: PropTypes.func.isRequired,
    onAdd: PropTypes.func.isRequired,
    onDelete: PropTypes.func.isRequired
  };

  static defaultProps = {};

  toggleRemoveAllTiles = () => {
    const { choices, onChange } = this.props;

    const allAtOne = this.allChoicesHaveCount(1);
    const update = choices.map(c => {
      c.categoryCount = allAtOne ? 0 : 1;
      return c;
    });
    onChange(update);
  };

  changeChoice = choice => {
    const { choices, onChange } = this.props;
    const index = choices.findIndex(h => h.id === choice.id);
    if (index !== -1) {
      choices.splice(index, 1, choice);
      onChange(choices);
    }
  };

  allChoicesHaveCount = count => {
    const { choices } = this.props;
    return every(choices, c => c.categoryCount === count);
  };

  render() {
    const {
      classes,
      className,
      choices,
      onAdd,
      onDelete,
      config,
      onConfigChange
    } = this.props;

    const categoryCountIsOne = this.allChoicesHaveCount(1);

    const choiceHolderStyle = {
      gridTemplateColumns: `repeat(${config.columns}, 1fr)`
    };
    return (
      <div className={classNames(classes.choices, className)}>
        <Header label="Choices" onAdd={onAdd} />

        <div className={classes.choiceHolder} style={choiceHolderStyle}>
          {choices.map((h, index) => (
            <Choice
              choice={h}
              correctResponseCount={h.correctResponseCount}
              key={index}
              onChange={this.changeChoice}
              onDelete={() => onDelete(h)}
            />
          ))}
        </div>
        <Config
          config={config}
          onChange={onConfigChange}
          categoryCountIsOne={categoryCountIsOne}
          onToggleCategoryCount={this.toggleRemoveAllTiles}
        />
      </div>
    );
  }
}
const styles = theme => ({
  row: {
    display: 'grid',
    gridColumnGap: `${theme.spacing.unit}px`,
    gridRowGap: `${theme.spacing.unit}px`,
    gridTemplateColumns: 'repeat(3, 1fr)',
    justifyContent: 'space-between',
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit
  },
  choiceHolder: {
    paddingTop: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    display: 'grid',
    gridRowGap: `${theme.spacing.unit}px`,
    gridColumnGap: `${theme.spacing.unit}px`
  },
  choices: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  label: {}
});
export default withStyles(styles)(Choices);
