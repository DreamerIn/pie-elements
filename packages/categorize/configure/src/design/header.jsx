import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import { AddButton } from './buttons';

export class Header extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    buttonLabel: PropTypes.string,
    onAdd: PropTypes.func.isRequired,
    label: PropTypes.string.isRequired
  };

  static defaultProps = {};
  render() {
    const { classes, className, onAdd, label, buttonLabel } = this.props;
    return (
      <div className={classNames(classes.header, className)}>
        <Typography variant="title" className={classes.title}>
          {label}
        </Typography>
        <AddButton onClick={onAdd} label={buttonLabel} />
      </div>
    );
  }
}
const styles = theme => ({
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: 2 * theme.spacing.unit
  }
});
export default withStyles(styles)(Header);
