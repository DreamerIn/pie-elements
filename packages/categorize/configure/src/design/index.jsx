import React from 'react';
import PropTypes from 'prop-types';
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';
import classNames from 'classnames';

export class Design extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string
  };
  static defaultProps = {};
  render() {
    const { classes, className } = this.props;
    return <div className={classNames(classes.design, className)} />;
  }
}
const styles = theme => ({
  design: {}
});
export default withStyles(styles)(Design);
