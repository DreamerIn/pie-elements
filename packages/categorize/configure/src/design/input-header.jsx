import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import EditableHtml from '@pie-lib/editable-html';

export class InputHeader extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    label: PropTypes.string,
    onChange: PropTypes.func,
    onDelete: PropTypes.func
  };

  static defaultProps = {};
  render() {
    const { onChange, label, classes, className } = this.props;
    return (
      <div className={classNames(classes.inputHeader, className)}>
        <EditableHtml
          autoWidthToolbar
          label={'label'}
          markup={label}
          onChange={onChange}
          className={classes.editor}
        />
      </div>
    );
  }
}
const styles = theme => ({
  editor: {
    flex: '1',
    paddingBottom: theme.spacing.unit
  },
  iconButtonRoot: {
    width: 'auto',
    height: 'auto'
  },
  inputHeader: {
    display: 'flex',
    justifyContent: 'space-between'
  }
});
export default withStyles(styles)(InputHeader);
