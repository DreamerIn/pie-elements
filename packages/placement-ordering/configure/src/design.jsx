import {
  FeedbackConfig,
  FormSection,
  InputCheckbox,
  InputContainer,
  TwoChoice
} from '@pie-lib/config-ui';
import EditableHtml from '@pie-lib/editable-html';

import { get, set } from 'nested-property';

import ChoiceEditor from './choice-editor';
import PropTypes from 'prop-types';
import React from 'react';
import TextField from 'material-ui/TextField';
import cloneDeep from 'lodash/cloneDeep';
import debug from 'debug';
import { withStyles } from 'material-ui/styles';

const log = debug('@pie-element:placement-ordering:design');

class Design extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      allMoveOnDrag: false
    };

    this.applyUpdate = modelFn => {
      const { model, onModelChange } = this.props;
      const update = modelFn(cloneDeep(model));
      onModelChange(update);
    };

    this.changeHandler = (modelPath, valuePath) => {
      return value => {
        log('[changeHandler] value: ', value);
        const v = valuePath ? get(value, valuePath) : value;
        this.applyUpdate(model => {
          set(model, modelPath, v);
          return model;
        });
      };
    };

    this.onLayoutChange = layout => {
      this.applyUpdate(model => {
        model.choiceAreaLayout = layout;
        return model;
      });
    };

    this.onPlacementTypeChange = event => {
      const includePlacment = event.currentTarget.checked;
      this.applyUpdate(model => {
        model.placementType = includePlacment ? 'placement' : 'none';
        return model;
      });
    };

    this.onDefaultLangChange = defaultLang => {
      this.applyUpdate(model => {
        model.defaultLang = defaultLang;
        return model;
      });
    };

    this.onPromptChange = this.changeHandler('prompt');
    this.onChoiceAreaLabelChange = this.changeHandler(
      'choiceAreaLabel',
      'target.value'
    );
    this.onAnswerAreaLabelChange = this.changeHandler(
      'answerAreaLabel',
      'target.value'
    );
    this.onFeedbackChange = this.changeHandler('feedback');
    this.onShuffleChange = this.changeHandler('shuffle', 'target.checked');
    this.onShowOrderingChange = this.changeHandler(
      'showOrdering',
      'target.checked'
    );

    this.onChoiceEditorChange = (choices, correctResponse) => {
      const { model, onModelChange } = this.props;
      const update = cloneDeep(model);
      update.choices = choices;
      update.correctResponse = correctResponse;
      onModelChange(update);
    };
  }

  render() {
    const { model, classes, imageSupport } = this.props;
    const { allMoveOnDrag } = this.state;
    return (
      <div className={classes.design}>
        <div className={classes.row}>
          <TwoChoice
            className={classes.orientation}
            header={'Orientation'}
            value={model.choiceAreaLayout}
            onChange={this.onLayoutChange}
            one={{ label: 'vertical', value: 'vertical' }}
            two={{ label: 'horizontal', value: 'horizontal' }}
          />
        </div>
        <div className={classes.row}>
          <InputCheckbox
            label="Shuffle"
            checked={model.shuffle}
            onChange={this.onShuffleChange}
            aria-label="shuffle"
          />
          <InputCheckbox
            label="Include placement area"
            checked={model.placementType === 'placement'}
            onChange={this.onPlacementTypeChange}
            aria-label="include-placment"
          />
          <InputCheckbox
            disabled={model.placementType !== 'placement'}
            label="Numbered guides"
            checked={model.showOrdering}
            onChange={this.onShowOrderingChange}
            aria-label="shuffle"
          />
        </div>
        <InputContainer label="Prompt" className={classes.promptHolder}>
          <EditableHtml
            className={classes.prompt}
            markup={model.prompt}
            onChange={this.onPromptChange}
            imageSupport={imageSupport}
          />
        </InputContainer>

        <div className={classes.row}>
          <TextField
            className={classes.choiceLabel}
            label="Choice label"
            value={model.choiceAreaLabel}
            onChange={this.onChoiceAreaLabelChange}
            fullWidth
          />
          {model.placementType === 'placement' && (
            <TextField
              label="Answer label"
              value={model.answerAreaLabel}
              onChange={this.onAnswerAreaLabelChange}
              fullWidth
            />
          )}
        </div>
        <FormSection label="Choices">
          <ChoiceEditor
            correctResponse={model.correctResponse}
            choices={model.choices}
            onChange={this.onChoiceEditorChange}
            imageSupport={imageSupport}
          />
        </FormSection>
        <FeedbackConfig
          feedback={model.feedback}
          onChange={this.onFeedbackChange}
          imageSupport={imageSupport}
        />
      </div>
    );
  }
}

Design.propTypes = {
  model: PropTypes.object.isRequired,
  onModelChange: PropTypes.func.isRequired
};

export default withStyles(theme => ({
  promptHolder: {
    width: '100%'
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit
  },
  row: {
    display: 'grid',
    gridAutoFlow: 'column',
    gridAutoColumns: '1fr',
    gridGap: '8px'
  },
  design: {
    paddingTop: '10px'
  },
  langControls: {
    marginTop: '0px',
    marginBottom: '0px'
  },
  choices: {
    marginTop: '20px',
    marginBottom: '20px'
  },
  orientation: {
    marginTop: '0px',
    marginBottom: '0px'
  }
}))(Design);
