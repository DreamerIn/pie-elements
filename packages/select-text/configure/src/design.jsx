import React from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';
import { withStyles } from 'material-ui/styles';
import cloneDeep from 'lodash/cloneDeep';
import { Tokenizer } from '@pie-lib/text-select';
import {
  InputCheckbox,
  InputContainer,
  NumberTextField,
  FeedbackConfig
} from '@pie-lib/config-ui';
import Chip from 'material-ui/Chip';
import debug from 'debug';

const log = debug('@pie-element:select-text:configure');

export class Design extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string
  };

  static defaultProps = {};

  changeText = event => {
    this.apply(u => {
      u.text = event.target.value;
      u.tokens = [];
    });
  };

  changeTokens = tokens => {
    this.apply(u => {
      u.tokens = tokens;
      const correctTokenCount = tokens.filter(t => t.correct).length;
      const max = isFinite(u.maxSelections) ? u.maxSelections : 0;
      u.maxSelections = Math.max(max, correctTokenCount);
    });
  };

  changeMaxSelections = (event, max) => {
    this.apply(u => (u.maxSelections = max));
  };

  changeHighlight = () => {
    this.apply(u => (u.highlightChoices = !u.highlightChoices));
  };

  apply = fn => {
    const { onChange, model } = this.props;
    const update = cloneDeep(model);
    fn(update);
    onChange(update);
  };

  changeFeedback = feedback => {
    this.apply(u => (u.feedback = feedback));
  };

  render() {
    const { model, classes, className } = this.props;

    log('[render] maxSelections:', model.maxSelections);

    return (
      <div className={className}>
        <TextField
          label="Content"
          className={classes.input}
          multiline
          value={model.text}
          onChange={this.changeText}
        />

        <InputCheckbox
          label={'Highlight choices'}
          checked={model.highlightChoices}
          onChange={this.changeHighlight}
        />

        <InputContainer label={'Tokens'} className={classes.tokenizerContainer}>
          <Tokenizer
            className={classes.tokenizer}
            text={model.text}
            tokens={model.tokens}
            onChange={this.changeTokens}
          />
        </InputContainer>
        <Chip
          label={`Selections Available: ${model.tokens.length}`}
          className={classes.chip}
        />
        <Chip
          label={`Correct Answers: ${
            model.tokens.filter(t => t.correct).length
          }`}
          className={classes.chip}
        />
        <NumberTextField
          min={0}
          label={'Selection count (0:any)'}
          max={model.tokens.length}
          value={model.maxSelections}
          onChange={this.changeMaxSelections}
        />

        <FeedbackConfig
          feedback={model.feedback}
          onChange={this.changeFeedback}
        />
      </div>
    );
  }
}
export default withStyles(theme => ({
  tokenizerContainer: {
    paddingRight: 0,
    marginRight: 0,
    '&:after': {
      position: 'absolute',
      bottom: 0,
      left: 0,
      right: 0,
      height: '1px',
      content: '""',
      backgroundColor: theme.palette.primary.main
    },
    marginBottom: theme.spacing.unit
  },
  chip: {
    marginRight: theme.spacing.unit * 2
  },
  input: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 3
  },
  tokenizer: {
    marginTop: theme.spacing.unit * 2
  },
  mainOpts: {
    width: '100%',
    justifyContent: 'space-between',
    display: 'flex',
    alignItems: 'baseline'
  }
}))(Design);
