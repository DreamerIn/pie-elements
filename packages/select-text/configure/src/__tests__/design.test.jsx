import { shallow } from 'enzyme';
import React from 'react';
import { Design } from '../design';

describe('design', () => {
  let w;
  let onChange;

  const getModel = () => ({ tokens: [] });
  beforeEach(() => {
    onChange = jest.fn();
    w = shallow(
      <Design
        model={getModel()}
        classes={{}}
        className={'foo'}
        onChange={onChange}
      />
    );
  });

  describe('snapshot', () => {
    it('renders', () => {
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    const assert = (fn, args, expected) => {
      const e = expected(getModel());
      it(`${fn} ${JSON.stringify(args)} => ${JSON.stringify(e)}`, () => {
        const instance = w.instance();
        instance[fn].apply(instance, args);
        expect(onChange).toBeCalledWith(e);
      });
    };

    describe('changeText', () => {
      assert('changeText', [{ target: { value: 'foo' } }], m => ({
        ...m,
        text: 'foo'
      }));
    });

    describe('changeTokens', () => {
      assert('changeTokens', [[{ start: 0, end: 1, text: 'f' }]], m => ({
        ...m,
        maxSelections: 0,
        tokens: [{ start: 0, end: 1, text: 'f' }]
      }));
      assert(
        'changeTokens',
        [[{ start: 0, end: 1, text: 'f', correct: true }]],
        m => ({
          ...m,
          maxSelections: 1,
          tokens: [{ start: 0, end: 1, text: 'f', correct: true }]
        })
      );
    });

    describe('changeMaxSelections', () => {
      assert('changeMaxSelections', [{}, 4], m => ({ ...m, maxSelections: 4 }));
    });
    describe('changeHighlight', () => {
      assert('changeHighlight', [], m => ({ ...m, highlightChoices: true }));
    });
    describe('changeFeedback', () => {
      assert('changeFeedback', [{ correctFeedbackType: 'none' }], m => ({
        ...m,
        feedback: { correctFeedbackType: 'none' }
      }));
    });
  });
});
