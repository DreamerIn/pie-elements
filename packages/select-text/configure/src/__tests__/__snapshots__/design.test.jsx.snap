// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design snapshot renders all items except feedback 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "correctAnswer": Object {
            "label": "Correct Answers",
            "settings": true,
          },
          "feedback": Object {
            "label": "Feedback",
            "settings": false,
          },
          "highlightChoices": Object {
            "label": "Highlight choices",
            "settings": true,
          },
          "mode": Object {
            "label": "Mode",
            "settings": true,
          },
          "partialScoring": Object {
            "label": "Allow Partial Scoring",
            "settings": false,
          },
          "prompt": Object {
            "label": "Prompt",
            "settings": true,
          },
          "rationale": Object {
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": false,
          },
          "selectionCount": Object {
            "label": "Selection count",
            "settings": true,
          },
          "selections": Object {
            "label": "Selections Available",
            "settings": true,
          },
          "studentInstructions": Object {
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "label": "Teacher Instructions",
            "settings": true,
          },
          "text": Object {
            "label": "Content",
            "settings": true,
          },
          "tokens": Object {
            "label": "Tokens",
            "settings": true,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "promptEnabled": undefined,
            "rationaleEnabled": undefined,
            "scoringType": false,
            "studentInstructionsEnabled": false,
            "teacherInstructionsEnabled": undefined,
          },
          "Settings": Object {
            "feedbackEnabled": false,
            "highlightChoices": undefined,
            "partialScoring": false,
          },
        }
      }
      model={
        Object {
          "tokens": Array [],
        }
      }
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <TextField
      label="Content"
      multiline={true}
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    />
    <Component
      label="Tokens"
    >
      <WithStyles(Tokenizer)
        onChange={[Function]}
        tokens={Array []}
      />
    </Component>
    <WithStyles(Chip)
      label="Mode: None"
    />
    <WithStyles(Chip)
      label="Selections Available: 0"
    />
    <WithStyles(Chip)
      label="Correct Answers: 0"
    />
    <Component
      label="Selection count (0:any)"
      max={0}
      min={0}
      onChange={[Function]}
    />
  </div>
</ConfigLayout>
`;

exports[`design snapshot renders all items except the content input 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "correctAnswer": Object {
            "label": "Correct Answers",
            "settings": true,
          },
          "feedback": Object {
            "label": "Feedback",
            "settings": false,
          },
          "highlightChoices": Object {
            "label": "Highlight choices",
            "settings": true,
          },
          "mode": Object {
            "label": "Mode",
            "settings": true,
          },
          "partialScoring": Object {
            "label": "Allow Partial Scoring",
            "settings": false,
          },
          "prompt": Object {
            "label": "Prompt",
            "settings": true,
          },
          "rationale": Object {
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": false,
          },
          "selectionCount": Object {
            "label": "Selection count",
            "settings": true,
          },
          "selections": Object {
            "label": "Selections Available",
            "settings": true,
          },
          "studentInstructions": Object {
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "label": "Teacher Instructions",
            "settings": true,
          },
          "text": Object {
            "label": "Content",
            "settings": false,
          },
          "tokens": Object {
            "label": "Tokens",
            "settings": true,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "promptEnabled": undefined,
            "rationaleEnabled": undefined,
            "scoringType": false,
            "studentInstructionsEnabled": false,
            "teacherInstructionsEnabled": undefined,
          },
          "Settings": Object {
            "feedbackEnabled": false,
            "highlightChoices": undefined,
            "partialScoring": false,
          },
        }
      }
      model={
        Object {
          "tokens": Array [],
        }
      }
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <Component
      label="Tokens"
    >
      <WithStyles(Tokenizer)
        onChange={[Function]}
        tokens={Array []}
      />
    </Component>
    <WithStyles(Chip)
      label="Mode: None"
    />
    <WithStyles(Chip)
      label="Selections Available: 0"
    />
    <WithStyles(Chip)
      label="Correct Answers: 0"
    />
    <Component
      label="Selection count (0:any)"
      max={0}
      min={0}
      onChange={[Function]}
    />
  </div>
</ConfigLayout>
`;

exports[`design snapshot renders all items with defaultProps 1`] = `
<ConfigLayout
  settings={
    <Panel
      configuration={
        Object {
          "correctAnswer": Object {
            "label": "Correct Answers",
            "settings": true,
          },
          "feedback": Object {
            "label": "Feedback",
            "settings": true,
          },
          "highlightChoices": Object {
            "label": "Highlight choices",
            "settings": true,
          },
          "mode": Object {
            "label": "Mode",
            "settings": true,
          },
          "partialScoring": Object {
            "label": "Allow Partial Scoring",
            "settings": false,
          },
          "prompt": Object {
            "label": "Prompt",
            "settings": true,
          },
          "rationale": Object {
            "label": "Rationale",
            "settings": true,
          },
          "scoringType": Object {
            "label": "Scoring Type",
            "settings": false,
          },
          "selectionCount": Object {
            "label": "Selection count",
            "settings": true,
          },
          "selections": Object {
            "label": "Selections Available",
            "settings": true,
          },
          "studentInstructions": Object {
            "label": "Student Instructions",
            "settings": false,
          },
          "teacherInstructions": Object {
            "label": "Teacher Instructions",
            "settings": true,
          },
          "text": Object {
            "label": "Content",
            "settings": true,
          },
          "tokens": Object {
            "label": "Tokens",
            "settings": true,
          },
        }
      }
      groups={
        Object {
          "Properties": Object {
            "promptEnabled": undefined,
            "rationaleEnabled": undefined,
            "scoringType": false,
            "studentInstructionsEnabled": false,
            "teacherInstructionsEnabled": undefined,
          },
          "Settings": Object {
            "feedbackEnabled": undefined,
            "highlightChoices": undefined,
            "partialScoring": false,
          },
        }
      }
      model={
        Object {
          "tokens": Array [],
        }
      }
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <TextField
      label="Content"
      multiline={true}
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    />
    <Component
      label="Tokens"
    >
      <WithStyles(Tokenizer)
        onChange={[Function]}
        tokens={Array []}
      />
    </Component>
    <WithStyles(Chip)
      label="Mode: None"
    />
    <WithStyles(Chip)
      label="Selections Available: 0"
    />
    <WithStyles(Chip)
      label="Correct Answers: 0"
    />
    <Component
      label="Selection count (0:any)"
      max={0}
      min={0}
      onChange={[Function]}
    />
  </div>
</ConfigLayout>
`;
